eth/downloader/downloader.go

```diff
@@ -432,27 +403,19 @@ func (d *Downloader) synchronise(id string, hash common.Hash, td, ttd *big.Int,
 	// Atomically set the requested sync mode
 	d.mode.Store(uint32(mode))
 
-	// Retrieve the origin peer and initiate the downloading process
-	var p *peerConnection
-	if !beaconMode { // Beacon mode doesn't need a peer to sync from
-		p = d.peers.Peer(id)
-		if p == nil {
-			return errUnknownPeer
-		}
-	}
 	if beaconPing != nil {
 		close(beaconPing)
 	}
-	return d.syncWithPeer(p, hash, td, ttd, beaconMode)
+	return d.syncToHead(hash)
 }
 
 func (d *Downloader) getMode() SyncMode {
 	return SyncMode(d.mode.Load())
 }
 
-// syncWithPeer starts a block synchronization based on the hash chain from the
-// specified peer and head hash.
-func (d *Downloader) syncWithPeer(p *peerConnection, hash common.Hash, td, ttd *big.Int, beaconMode bool) (err error) {
```

Comment:
It seems like this method could (ought to be?) be merged into `synchronise`.  However, it will create a monster method, so I opt to keep it separate for now.